### PR TITLE
[nvml] Initial commit for Non-Volatile Memory Library

### PR DESCRIPTION
**Pull request by krzycz**

### CODE REVIEW

File: ports/nvml/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    
+# Install libraries (debug)
+file(GLOB LIB_DEBUG_FILES ${DEBUG_ARTIFACTS_PATH}/lib[pv]mem*.lib ${DEBUG_ARTIFACTS_PATH}/lib[pv]mem*.exp)
+file(INSTALL ${LIB_DEBUG_FILES} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib)
+file(REMOVE ${CURRENT_PACKAGES_DIR}/debug/lib/libpmemcommon.lib)
+file(GLOB LIB_DEBUG_FILES ${DEBUG_ARTIFACTS_PATH}/lib[pv]mem*.dll)
+file(INSTALL ${LIB_DEBUG_FILES} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin)
+
+# Install tools (debug)
+file(INSTALL ${DEBUG_ARTIFACTS_PATH}/pmempool.exe DESTINATION ${CURRENT_PACKAGES_DIR}/debug/tools)
```
Comment (ras0219-msft) [inline ports/nvml/portfile.cmake]: We typically don't install debug versions of tools.

File: ports/nvml/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+# Install libraries (release)
+file(GLOB LIB_RELEASE_FILES ${RELEASE_ARTIFACTS_PATH}/lib[pv]mem*.lib ${RELEASE_ARTIFACTS_PATH}/lib[pv]mem*.exp)
+file(INSTALL ${LIB_RELEASE_FILES} DESTINATION ${CURRENT_PACKAGES_DIR}/lib)
+file(REMOVE ${CURRENT_PACKAGES_DIR}/lib/libpmemcommon.lib)
+file(GLOB LIB_RELEASE_FILES ${RELEASE_ARTIFACTS_PATH}/lib[pv]mem*.dll)
+file(INSTALL ${LIB_RELEASE_FILES} DESTINATION ${CURRENT_PACKAGES_DIR}/bin)
+
+# Install tools (release)
+file(INSTALL ${RELEASE_ARTIFACTS_PATH}/pmempool.exe DESTINATION ${CURRENT_PACKAGES_DIR}/tools)
```
Comment (ras0219-msft) [inline ports/nvml/portfile.cmake]: This should go into a `nvml` subdirectory of tools and then you should also call `vcpkg_copy_tool_dependencies(${CURRENT_PACKAGES_DIR}/tools/nvml)` to ensure any runtime DLLs are deployed applocally to it.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Overall looks great!

I had a few comments about tools and I'll be happy to merge once those are addressed.

File: ports/nvml/portfile.cmake (hunk 2 above)
Comment (krzycz) [inline ports/nvml/portfile.cmake]: Fixed.

File: ports/nvml/portfile.cmake (hunk 1 above)
Comment (krzycz) [inline ports/nvml/portfile.cmake]: Fixed.

**Comment by ras0219-msft** [comment]
Hi @krzycz, did you sign the CLA? I need that to merge this PR.

**Review by ras0219-msft — APPROVED** [review]

**Comment by krzycz** [comment]
@ras0219-msft - CLA signed

**Comment by ras0219-msft** [comment]
Thanks!

**Pull request merged**